### PR TITLE
various enhancements

### PR DESCRIPTION
**Pull request by gregcaporaso**
* Adds ``info`` command which provides ``print_qiime_config.py``-like information
* Adds ``tools`` group which contains ``extract`` and ``view`` (#17) (and will contain ``import`` soon)
* Adds method/visualizer name and description as short_help and help, respectively (#16)
* Removes ``--plugins`` and ``--info`` parameters, as this functionality is now accessible in ``qiime info``.

### CODE REVIEW

File: q2cli/_tools.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        index_extension = index_extension[1:]
+    try:
+        type_ = qiime.sdk.Result.peek(visualization_path).type
+    except zipfile.BadZipFile:
+        raise click.BadParameter(
+            '%s is not a QIIME Result. Only QIIME Visualizations can be '
+            'viewed.' % visualization_path)
+    # TODO: is there a better way to do this type check (without accessing
+    # qiime.core?)
+    if type_ != qiime.core.type.Visualization:
```
Comment (gregcaporaso) [inline q2cli/_tools.py]: is there a better way to do this type check (without accessing ``qiime.core``)?

Comment (ebolyen) [inline q2cli/_tools.py]: Not at the moment, we might think about exposing types at something like `qiime.sdk.types` (in fact we would probably move `semantic.py`, `primitive.py` and `visualizations.py`, leaving `grammar.py` in core), but I think we need a larger conversation about package structure post-alpha.

File: q2cli/_info.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+def _echo_plugins():
+    plugin_manager = qiime.sdk.PluginManager()
+    installed_plugins = plugin_manager.plugins
+    if len(installed_plugins) == 0:
+        # TODO: update this URL, pending:
+        # https://github.com/qiime2/qiime2/issues/80
+        # https://github.com/qiime2/qiime2/issues/82
+        click.secho('No plugins are currently installed.\nYou can browse '
+                    'the official QIIME 2 plugins at: '
+                    'https://github.com/qiime2.')
```
Comment (ebolyen) [inline q2cli/_info.py]: no period at the end of the URL

File: q2cli/_info.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import sys
+
+import pip
+import click
+import qiime
+
+from . import __version__ as q2cli_version
+
+
+def _echo_version(ctx, name, value):
```
Comment (ebolyen) [inline q2cli/_info.py]: make these optional with a default `None`

File: q2cli/_info.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        click.echo(e)
+
+
+@click.command(help='Display information about QIIME.')
+@click.option('--py-packages', is_flag=True,
+              help='Display names and versions of all installed Python '
+                   'packages.')
+def info(py_packages):
+    click.secho('System versions', fg='green')
+    _echo_version(None, None, True)
```
Comment (ebolyen) [inline q2cli/_info.py]: This is gross, make them optional

File: q2cli/_tools.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    visualization = qiime.sdk.Visualization.load(visualization_path)
+    index_paths = visualization.get_index_paths()
+    if index_extension not in index_paths:
+        raise click.BadParameter(
+            'No index %s file with is present in the archive. Available index '
+            'extensions are: %s' % (index_extension,
+                                    ', '.join(index_paths.keys())))
+    else:
+        # TODO: we're currently extracting twice here - once when we load, and
+        # again here. How can this be avoided?
```
Comment (ebolyen) [inline q2cli/_tools.py]: use `visualization.get_index_paths(relative=False)` instead of extracting.

File: q2cli/_tools.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+                # Entering either ^C or Return will cause output_dir to be
+                # cleaned up.
+                click.prompt('Press the Return key or ^C to quit (your '
+                             'visualization may no longer be accessible)',
+                             default='q', prompt_suffix='. ',
+                             show_default=False)
+
+
+@tools.command(help='Extract a QIIME Arifact or Visualization.')
+@click.argument('path')
```
Comment (ebolyen) [inline q2cli/_tools.py]: :+1: for positional arg

File: q2cli/_tools.py [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+                    "the command exits, use 'qiime extract'.")
+@click.argument('visualization-path')
+@click.option('--index-extension', required=False, default='html',
+              help='The extension of the index file that should be opened. '
+                   '[default: html]')
+def view(visualization_path, index_extension):
+    if index_extension.startswith('.'):
+        index_extension = index_extension[1:]
+    try:
+        type_ = qiime.sdk.Result.peek(visualization_path).type
```
Comment (ebolyen) [inline q2cli/_tools.py]: You could use `qiime.sdk.Visualization.peek` instead, it would error if it's an `Artifact`.

File: q2cli/cli.py [hunk 8]
```diff
@@ -6,30 +6,35 @@
...
+    _plugin_manager = PluginManager()
+    _builtin_commands = {'tools': q2cli._tools.tools,
+                         'info': q2cli._info.info}
+
     def list_commands(self, ctx):
-        plugins = PLUGIN_MANAGER.plugins.keys()
-        return sorted(plugins)
+        plugins = list(sorted(self._plugin_manager.plugins.keys()))
+        commands = list(sorted(self._builtin_commands.keys())) + plugins
+        return commands
```
Comment (ebolyen) [inline q2cli/cli.py]: The `+ plugins` is kind of hiding at the end, what if you did:
`return builtins + plugins` which would be more obvious.

**Comment by ebolyen** [comment]
Done reviewing, this is pretty awesome! Couple of places where you could simplify the code, especially using `get_index_paths(relative=False)`

File: q2cli/_tools.py (hunk 7 above)
Comment (gregcaporaso) [inline q2cli/_tools.py]: Awesome, that lets me get around accessing ``qiime.core``. Note that the ``except (zipfile.BadZipFile, KeyError)`` catches passing a non-zip or non-result file.

File: q2cli/_tools.py (hunk 5 above)
Comment (gregcaporaso) [inline q2cli/_tools.py]: 🎸

File: q2cli/_tools.py [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+        raise click.BadParameter(
+            '%s is not a QIIME Visualization. Only QIIME Visualizations can '
+            'be viewed.' % visualization_path)
+    except TypeError:
+        raise click.BadParameter(
+            '%s is not a Visualization. Only Visualizations can be viewed.' %
+            visualization_path)
+
+    visualization = qiime.sdk.Visualization.load(visualization_path)
+    index_paths = visualization.get_index_paths(relative=False)
```
Comment (ebolyen) [inline q2cli/_tools.py]: :+1:

**Comment by ebolyen** [comment]
This is awesome!

**Pull request merged**